packages/repository/src/model.ts

```diff
@@ -106,6 +106,17 @@ export class ModelDefinition {
     const definition = (definitionOrType as PropertyDefinition).type
       ? (definitionOrType as PropertyDefinition)
       : {type: definitionOrType};
+
+    if (
+      definition.id === true &&
+      definition.generated === true &&
+      definition.type === 'string'
```

Comment:
I am afraid we misunderstood each other. Here is what I meant:

```suggestion
      definition.type !== undefined
```

Either way, please add a new test to verify that `useDefaultIdType` is not set when `definition.type` was not provided (and was not inferred from TypeScript metadata either).

Reply:
There is no model in the tests of the type as undefined, how should I generate a model with these features?